File: Models/TabModel.cs
using System.Collections.Generic;
using System.Dynamic;

namespace EmployeeViewer.Models
{
     public class TabModel
     {
          public int tabID { get; set; }
          public string displayName { get; set; }
          public string name { get; set; }
          public List<FieldModel> fields { get; set; }
          public List<ExpandoObject> data { get; set; }
     }

     public class DataModel
     {
          public int tabID { get; set; }
          public string values { get; set; }
     }
}

File: ClientApp/src/app/dynamic/models/tab.model.ts
import { FieldModel } from "./field.model";

export class TabModel {
  tabID: number;
  displayName: string;
  name: string;
  fields: FieldModel[];
  data: any;
}

export class DataModel {
  tabID: number;
  values: string;
}

export class TabDataModel {
  values: any;
  view: TabModel;
}
